### PR TITLE
EMBCESSMOD-1571 - Security Phrase Tab, Navigation Improvements

### PR DESCRIPTION
**Pull request by MShayOffice**
Set up and finished Security Phrase tab
Fixed "scroll to top" for changing tabs or steps
Adjusted some positioning for Security Questions tab
tabItem.route is now consistently used when checking if route can be accessed (review tab)
Removed some unused commented-out code

### CODE REVIEW

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/ess-file-components/security-phrase/security-phrase.component.html [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            length and does not include any special characters. You may include
+            numbers and letters. The security phrase is not case sensitive.
+          </p>
+
+          <p>
+            For security reasons, your security phrase should not include your
+            email address.
+          </p>
+
+          <div class="row">
```
Comment (SodhiA1) [inline responders/src/UI/embc-responder/src/app/feature-components/wizard/ess-file-components/security-phrase/security-phrase.component.html:31]: Use form group

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/ess-file-components/security-phrase/security-phrase.component.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+  constructor(
+    private stepCreateEssFileService: StepCreateEssFileService,
+    private customValidationService: CustomValidationService,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    // Set up form validation for verification check
+    this.securityPhraseFC = new FormControl(
```
Comment (SodhiA1) [inline responders/src/UI/embc-responder/src/app/feature-components/wizard/ess-file-components/security-phrase/security-phrase.component.ts:29]: Use form group to enclose both securityPhraseFC and bypassPhrase controls

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html [hunk 3]
```diff
@@ -22,7 +22,7 @@
         'tab-margin custom-tab-link': tabItem.name !== 'review'
       }"
     >
-      <span class="span-height" (click)="isAllowed(tabItem.name, $event)"
+      <span class="span-height" (click)="isAllowed(tabItem.route, $event)"
```
Comment (SodhiA1) [inline responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html:25]: 'route' is the incorrect property to be used here. The correct property that should be used is 'name'

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-profile/step-create-profile.component.html [hunk 4]
```diff
@@ -22,7 +22,7 @@
         'tab-margin custom-tab-link': tabItem.name !== 'review'
       }"
     >
-      <span class="span-height" (click)="isAllowed(tabItem.name, $event)"
+      <span class="span-height" (click)="isAllowed(tabItem.route, $event)"
```
Comment (SodhiA1) [inline responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-profile/step-create-profile.component.html:25]: 'route' is the incorrect property to be used here. The correct property that should be used is 'name'

**Review by SodhiA1 — CHANGES_REQUESTED** [review]

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html (hunk 3 above)
Comment (MShayOffice) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html:25]: Then why is the property called tabRoute in the isAllowed method?

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/ess-file-components/security-phrase/security-phrase.component.ts (hunk 2 above)
Comment (MShayOffice) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/ess-file-components/security-phrase/security-phrase.component.ts:29]: BypassPhrase has to be separated from SecurityPhrase, the two are mutually exclusive. If they're grouped, we'd have to add a bunch of extra code to make them not act like a group.

Comment (SodhiA1) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/ess-file-components/security-phrase/security-phrase.component.ts:29]: Please add conditional validation to control the mutually exclusive behavior. All user input fields needs to be part of a form. I just checked profile/SecurityQuestionsComponent, :
this.bypassQuestions = new FormControl(
      this.stepCreateProfileService.bypassSecurityQuestions
    );
this needs to be moved to formgroup as well. My bad, I missed it during the review of this component.

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html (hunk 3 above)
Comment (SodhiA1) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html:25]: Routes can change, name property is more reliable should be used at both tab and step level. These changes are WIP, are you using this function somewhere?

Comment (MShayOffice) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html:25]: IsAllowed is used when checking if the Review tab can be accessed. If we're consistently passing the name into it, then the first parameter in that function should be changed to tabName or tabRouteName.

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/ess-file-components/security-phrase/security-phrase.component.ts (hunk 2 above)
Comment (MShayOffice) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/ess-file-components/security-phrase/security-phrase.component.ts:29]: I don't like this, but for solidarity with the rest of the code, I'll group the loose FormControls on Security Phrase, Security Question, and Review & Save.

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html (hunk 3 above)
Comment (SodhiA1) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html:25]: Please leave it as is, this is WIP, I am working on it.

Comment (MShayOffice) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/step-create-ess-file/step-create-ess-file.component.html:25]: Okay, didn't realize you were.

**Review by SodhiA1 — APPROVED** [review]

**Pull request merged**